src/pygama/lgdo/vectorofvectors.py

```diff
@@ -55,35 +59,48 @@ def __init__(
             of `flattened_data` if it was not supplied.
         dtype
             Sets the type of data stored in `flattened_data`. Required if
-            `flattened_data` is ``None``.
+            `flattened_data` and `listoflists` are ``None``.
         attrs
             A set of user attributes to be carried along with this LGDO.
         """
-        if cumulative_length is None:
-            self.cumulative_length = Array(
-                shape=(shape_guess[0],), dtype="uint32", fill_val=0
+        if listoflists is not None:
+            cum_length = []
+            for vec in listoflists:
+                cum_length.append(len(vec))
+            self.flattened_data = Array(
+                nda=np.array([item for sublist in listoflists for item in sublist])
```

Comment:
Python for loops with built-in lists... Can this be made more efficient with Numpy?

Reply:
So the issue is that numpy doesn't like jagged arrays. I'm not sure what an efficient way around that would be.

Reply:
I think I found a more efficient way. It should be good enough for us.

Reply:
In the long term we'll want to have more interoperability between `VectorOfVectors` and [`awkward.Array`](https://awkward-array.org/doc/main/index.html). Once #377 is addressed, we could support passing an awkward array as `listoflists`, that should be the fastest.